scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java

```diff
@@ -87,7 +87,7 @@ public void register(NotificationRequest notificationRequest) throws Notificatio
         DateTime nextStartTime = request.getStartTime();
         DateTime endTime;
         if (request.getEndTime().isBefore(currentTime)) {
-            endTime = request.getEndTime();
+            endTime = request.getEndTime().minusMinutes(1);
```

Comment:
Should we minus the minute after if-else? Also, this change is causing some UT failures, please fix those.